### PR TITLE
[WIP] Adds cloth and leather

### PR DESCRIPTION
**Pull request by unid15**
They were in the game before but they had absolutely no use. Now they do (or at least cloth does)

* Cloth can be used to make rags (1 roll -> 2 rags) as well as some basic clothing items:

  * Turbans [(](http://puu.sh/szAO1/6f77eb2bed.jpg)2 rolls) for the head slot
  * Hand and foot wraps (4 rolls) for hand/feet slots respectively
  * Togas (12 rolls) for uniform slot
  * Roughspun robes (12 rolls) for suit slot (they were already in the game)

 - wraps
 - togas

* Cloth can also be used to wrap up bleeding wounds. They're less effective than bandages and gauze, and it takes 6 seconds to apply it.

* Using sharp items on clothing items lets you destroy them to salvage cloth and leather (depending on the material). When destroying cloth items, some of the cloth may be turned into useless for crafting rags - but that mostly happens when you use crappy and dull tools.

  * Destruction takes from 10 to 5 seconds, depending on tools used. Glass shards are slower, knives are faster.

  * Gloves usually turn into 2 sheets, hats and masks into 1, shoes into 2, suits and clothing into 10.

  * Some items can't be destroyed, including reinforced and armored items, as well as head of staff uniforms.

currently leather is useless, but leather clothing is much less common than cloth

### CODE REVIEW

**Comment by yclatious** [comment]
More ways to gheto fix your problems are fine by me, how do you even get cloth, butchering?

**Comment by Probe1** [comment]
sure why not

**Comment by JMWTurner** [comment]
@yclatious suggestions:
* From plant thread (would have to add some plants that can be harvested for fibers. Or add that property to kudzu if cut with plant clippers)
* From animal yarn (shaving space bears why not)
* From giant spider silk (maybe by using a metal rod on webs)

**Comment by Exxion** [comment]
Can you cut up bedsheets into cloth, or still only into rags?

**Comment by pvak** [comment]
can you also make a special autolathe that uses cloth and leather to make all kinds of cool clothes?

**Comment by Ephx1** [comment]
> Headwraps

Ohhhhhh dear, rev up those unfunny meme muslim chaplains

But seriously, I'm excited for cloth bandages.
Could you maybe do like cloth slings for broken bones? I've seen body casts used once, by me, because I had it as a borg, and the guy didn't want to bother with surgery. It'd be neat to have as a ghetto version, but hell, since they're so underused anyways it'd be fine if they worked like the real thing.

**Comment by yclatious** [comment]
Id like to suggest that you can have a physical method of getting them, like a cotton plant, and a way to synth it from the bio fabricator Botany has.

That way, ghetto botany can be a makeshift first aid station and help people out a bit.

**Comment by Chronakai** [comment]
Would the recycler/furnace rip up backpacks, jumpsuits, and satchels into cloth or leather as well?

**Comment by Shadowmech88** [comment]
Now we just need a sewing machine to allow players to craft more complex clothing items.
Maybe have it use a blueprint system like the mechanics' general fabricator does, so you have to have an item on-hand if you want to make more using cloth.

File: code/modules/mining/materials.dm [hunk 1]
```diff
@@ -91,20 +91,41 @@ var/global/list/initial_materials	//Stores all the matids = 0 in helping New
 		return
 	addAmount(mat_id,-amount)
 
+/datum/materials/proc/removeSheetAmount(mat_id, amount)
```
Comment (ComicIronic) [inline code/modules/mining/materials.dm:94]: Consider using "number" instead of "amount" - in the rest of the materials code, amount is used for CCs.

**Comment by ComicIronic** [comment]
Heavily oppose a turban being added - it's an item with nothing but meme potential in any playerbase.

**Comment by SonixApache** [comment]
@ComicIronic Did you know that there are more types of turban-wearing people than the ones with AKs and the ones that blow themselves up?

**Comment by ComicIronic** [comment]
There are also more people who shit in toilets than who shit in their hands and smear it across a wall, but tell that to #3000. I don't support adding content which is obviously going to be exploited by the worst level of player.

**Comment by Ephx1** [comment]
@SonixApache
And _how_ many times have you seen our players play such a character let alone role play seriously in general?

Get real, the only people who are going to wear it are bored assistants and chaplains who haven't gotten the memo that playing that obnoxious gimmick is terrible.

**Comment by Exxion** [comment]
I'd have no problem with it if this were anywhere except here
But knowing our playerbase, this wouldn't go well

**Comment by unid15** [comment]
Will reopen later

**Pull request closed without merging**